### PR TITLE
updated recurrence.md which fixes #1564

### LINKED ISSUE #1564: Recurrent docs need to be update for v0.12
The current doc examples for recurrent layers all use `rand` or `randn` directly instead of `rand(Float32, ...)`. This leads to a method error since we enforced a `eltype` matching in v0.12.

### PR DESCRIPTION
**Pull request by aditkumar72**
changed `rand` and `randn` definition to `rand(Float32, ...)` and `randn(Float32, ...)`

### CODE REVIEW

**Comment by mcabbott** [comment]
Note that if you try to run the examples, you get
```
julia> rnn = Flux.RNNCell(2, 5)
ERROR: invalid redefinition of constant rnn
```
from this line https://github.com/FluxML/Flux.jl/pull/1637/files#diff-2a4b8a2ddf993f91e23b12268baaeb5e5c9760950a9806e3a87fef56c96bfa92R40. Maybe either the first or the second `rnn` should have a different name.

File: docs/src/models/recurrence.md [hunk 1]
```diff
@@ -13,17 +13,17 @@ An aspect to recognize is that in such model, the recurrent cells `A` all refer
...
 
 ```julia
-Wxh = randn(5, 2)
-Whh = randn(5, 5)
-b   = randn(5)
+Wxh = randn(Float32, 5, 2)
+Whh = randn(Float32, 5, 5)
+b   = randn(Float32, 5)
 
 function rnn(h, x)
```
Comment (darsnack) [inline docs/src/models/recurrence.md:20]:
```suggestion
 function rnn_cell(h, x)
```

File: docs/src/models/recurrence.md [hunk 2]
```diff
@@ -13,17 +13,17 @@ An aspect to recognize is that in such model, the recurrent cells `A` all refer
...
   h = tanh.(Wxh * x .+ Whh * h .+ b)
   return h, h
 end
```
Comment (darsnack) [inline docs/src/models/recurrence.md:23]: Can you fix the weird one space indent here?

**Review by darsnack — CHANGES_REQUESTED** [review]
Small changes, but looks good

**Comment by aditkumar72** [comment]
@darsnack I have made the changes can you please review

**Review by darsnack — APPROVED** [review]
Thanks, looks great!

**Comment by aditkumar72** [comment]
Thank you for reviewing and approving my PR.

**Pull request merged**